ir/memory.cpp

```diff
@@ -431,6 +431,36 @@ Memory::Memory(State &state, bool little_endian)
   local_block_liveness
     = expr::mkLambda({ expr::mkVar("#bid0", bitsBid() - 1) }, false);
 
+  // The first byte of the memory cannot be allocated because it is for a null
+  // pointer.
+  // The last byte of the memory cannot be allocated because ptr + size
+  // (where size is the size of the block and ptr is the beginning address of
+  // the block) should not overflow.
+  avail_space = expr::mkUInt(-2, bitsPtrSize());
```

Comment:
this should be a variable

Reply:
Is it performance reason or correctness reason? I think members like `local_blk_addr` are also expressions rather than Z3 variables.

Reply:
I made it as a variable so initially arbitrary number of bytes are in use